client_test.go

```diff
@@ -37,3 +40,21 @@ func newMockedClient(t *testing.T, requestMockFile string, statusCode int) *http
 		return resp
 	})
 }
+
+func TestRateLimitRetryError(t *testing.T) {
+	c := newTestClient(func(*http.Request) *http.Response {
+		return &http.Response{
+			StatusCode: http.StatusTooManyRequests,
+			Header:     http.Header{"Retry-After": []string{"0"}},
+		}
+	})
+	client := notionapi.NewClient("some_token", notionapi.WithHTTPClient(c))
+	_, err := client.Block.Get(context.Background(), notionapi.BlockID("some_block_id"))
+	if err == nil {
+		t.Errorf("Get() error = %v", err)
+	}
+	wantErr := "Retry request with 429 response failed"
+	if err.Error() != wantErr {
+		t.Errorf("Get() error = %v, wantErr %s", err, wantErr)
+	}
```

Comment:
Nitpick: it might be a good thing to test that we actually retried the correct number of times.